Sources/Tokenizers/PreTokenizer.swift

```diff
@@ -9,27 +9,26 @@ import Foundation
 import Hub
 
 public protocol PreTokenizer {
-    func preTokenize(text: String) -> [String]
-    func preTokenize(texts: [String]) -> [String]
-    func callAsFunction(texts: [String]) -> [String]
-    func callAsFunction(text: String) -> [String]
+    func preTokenize(text: String, firstSection: Bool) -> [String]
+    func preTokenize(texts: [String], firstSection: Bool) -> [String]
+    func callAsFunction(texts: [String], firstSection: Bool) -> [String]
+    func callAsFunction(text: String, firstSection: Bool) -> [String]
```

Comment:
Unfortunately I had to change this public protocol so that `MetaspacePreTokenizer` can conditionally work depending on whether we are handling the first split. This is necessary because the input text is split by the added tokens first, and then all the chunks go through the tokenization pipeline. Do you think this will break something in your codebases @ZachNagengast @awni?

Reply:
@davidkoski wdyt?

Reply:
A default value is provided in a protocol extension, but the protocol definition itself cannot have default values, unfortunately.

Reply:
looks fine for mlx-swift.  I wonder if this should be an enum instead of a bool?  I don't know if there are other "contexts" that might be used...

Reply:
Good call @davidkoski, I did check the transformers.js port and didn't see any other context options, but better to play it safe.